File: platform/wlmedia_hm/libwlmedia/src/main/ets/com/ywl5320/wlmedia/bean/WlMediaInfoBean.ts
import { WlTrackInfoBean } from "./WlTrackInfoBean";

/**
 * author : ywl5320
 * e-mail : ywl5320@163.com
 * desc   : wlmedia
 * date   : 2025/1/10
 */
export class WlMediaInfoBean {
  duration: number;
  startTime: number;
  bitRate: number;

  audioTracks: WlTrackInfoBean[] | undefined;
  videoTracks: WlTrackInfoBean[] | undefined;
  subtitleTracks: WlTrackInfoBean[] | undefined;

  constructor() {
  }
}

File: wlmedia_android/wlmedia/src/main/java/com/ywl5320/wlmedia/bean/WlMediaInfoBean.java
package com.ywl5320.wlmedia.bean;

/**
 * author : ywl5320
 * e-mail : ywl5320@163.com
 * desc   : wlmedia
 * date   : 2024/8/11
 */
public class WlMediaInfoBean {
    private double duration;
    private double startTime;
    private long bitRate;

    private WlTrackInfoBean[] audioTracks;
    private WlTrackInfoBean[] videoTracks;
    private WlTrackInfoBean[] subtitleTracks;

    public WlMediaInfoBean() {
    }

    public double getDuration() {
        return duration;
    }

    public double getStartTime() {
        return startTime;
    }

    public long getBitRate() {
        return bitRate;
    }

    public WlTrackInfoBean[] getAudioTracks() {
        return audioTracks;
    }

    public WlTrackInfoBean[] getVideoTracks() {
        return videoTracks;
    }

    public WlTrackInfoBean[] getSubtitleTracks() {
        return subtitleTracks;
    }
}
